File: lib_common/src/main/java/com/fly/tour/common/util/AnimationUtils.java
package com.zjx.vcars.common.util;

import android.view.animation.Animation;
import android.view.animation.TranslateAnimation;

/**
 * @Description
 * @Author guojingbu
 * @Date 2019/3/12
 */
public class AnimationUtils {
    /**
     * 从控件所在位置移动到控件的底部
     *
     * @return
     */
    public static TranslateAnimation moveToBottom() {

        TranslateAnimation hiddenAction = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 0.0f,
                Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 1.0f);
        hiddenAction.setDuration(300);
        return hiddenAction;
    }

    public static TranslateAnimation moveToLocation() {
        TranslateAnimation visibleAction = new TranslateAnimation(Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 0.0f,
                Animation.RELATIVE_TO_SELF, 1.0f, Animation.RELATIVE_TO_SELF, 0.0f);
        visibleAction.setDuration(300);
        return visibleAction;

    }
}

File: lib_common/src/main/java/com/fly/tour/common/util/AnimationUtils.kt
package com.fly.tour.common.util

import android.view.animation.Animation
import android.view.animation.TranslateAnimation

/**
 * @Description
 * @Author guojingbu
 * @Date 2019/3/12
 */
object AnimationUtils {
    /**
     * 从控件所在位置移动到控件的底部
     *
     * @return
     */
    fun moveToBottom(): TranslateAnimation {

        val hiddenAction = TranslateAnimation(Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 0.0f,
                Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 1.0f)
        hiddenAction.duration = 300
        return hiddenAction
    }

    fun moveToLocation(): TranslateAnimation {
        val visibleAction = TranslateAnimation(Animation.RELATIVE_TO_SELF, 0.0f, Animation.RELATIVE_TO_SELF, 0.0f,
                Animation.RELATIVE_TO_SELF, 1.0f, Animation.RELATIVE_TO_SELF, 0.0f)
        visibleAction.duration = 300
        return visibleAction

    }
}
